src/fheroes2/maps/map_format_helper.cpp

```diff
@@ -190,4 +193,235 @@ namespace Maps
 
         addObjectToTile( map.tiles[tileId], group, index, uid );
     }
+
+    void updateMapPlayers( Map_Format::MapFormat & map )
+    {
+        static_assert( Color::BLUE == 1 << 0, "The kingdom color values have changed. You are going to break map format!" );
+        static_assert( Color::GREEN == 1 << 1, "The kingdom color values have changed. You are going to break map format!" );
+        static_assert( Color::RED == 1 << 2, "The kingdom color values have changed. You are going to break map format!" );
+        static_assert( Color::YELLOW == 1 << 3, "The kingdom color values have changed. You are going to break map format!" );
+        static_assert( Color::ORANGE == 1 << 4, "The kingdom color values have changed. You are going to break map format!" );
+        static_assert( Color::PURPLE == 1 << 5, "The kingdom color values have changed. You are going to break map format!" );
+
+        static_assert( Race::NONE == 0, "The race values have changed. You are going to break map format!" );
+        static_assert( Race::KNGT == 1 << 0, "The race values have changed. You are going to break map format!" );
+        static_assert( Race::BARB == 1 << 1, "The race values have changed. You are going to break map format!" );
+        static_assert( Race::SORC == 1 << 2, "The race values have changed. You are going to break map format!" );
+        static_assert( Race::WRLK == 1 << 3, "The race values have changed. You are going to break map format!" );
+        static_assert( Race::WZRD == 1 << 4, "The race values have changed. You are going to break map format!" );
+        static_assert( Race::NECR == 1 << 5, "The race values have changed. You are going to break map format!" );
+        static_assert( Race::MULT == 1 << 6, "The race values have changed. You are going to break map format!" );
+        static_assert( Race::RAND == 1 << 7, "The race values have changed. You are going to break map format!" );
+
+        const size_t mainColors{ 6 };
```

Comment:
I suggest to make it `constexpr` instead of `const`. While `const` variables of basic types may be initialized at compile time, this is not guaranteed, because formally it's just an optional compiler optimization.

```suggestion
        constexpr size_t mainColors{ 6 };
```

Reply:
Also, there is `KINGDOMMAX` define in `gamedefs.h` that is used to define maximum number of kingdoms/players in the game.